include/meltpooldg/reinitialization/reinitialization_problem.hpp

```diff
@@ -143,6 +117,148 @@ namespace Reinitialization
       reinit_operation.initialize(scratch_data, solution_level_set, base_in->parameters);
     }
 
+    void 
+    setup_dof_system( std::shared_ptr<SimulationBase<dim>> base_in,
+                      const bool do_initial_setup )
+    {
+      /*
+       *  setup scratch data
+       */
+      if (do_initial_setup)
+      {
+        scratch_data = std::make_shared<ScratchData<dim>>();
+        /*
+         *  setup mapping
+         */
+        const auto mapping = MappingQGeneric<dim>(base_in->parameters.base.degree);
+        scratch_data->set_mapping(mapping); 
+        /*
+         *  create quadrature rule
+         */
+        QGauss<1> quad_1d_temp(base_in->parameters.base.n_q_points_1d);
+        scratch_data->attach_quadrature(quad_1d_temp); 
+      }
+      /*
+       *  setup DoFHandler
+       */
+      FE_Q<dim>    fe(base_in->parameters.base.degree); 
+      dof_handler.initialize(*base_in->triangulation, fe );
+      
+      if (do_initial_setup)
+        scratch_data->attach_dof_handler(dof_handler);
+      /*
+       *  re-create partitioning
+       */
+      scratch_data->create_partitioning();
+      /*
+       *  make hanging nodes constraints
+       */
+      constraints.clear();
+      constraints.reinit(scratch_data->get_locally_relevant_dofs());
+      DoFTools::make_hanging_node_constraints(dof_handler, constraints);
+      constraints.close();
+      
+      if (do_initial_setup)
+        scratch_data->attach_constraint_matrix(constraints);
+      
+      /*
+       *  create the matrix-free object
+       */
+      scratch_data->build();
+    }
+    
+    /*
+     *  perform mesh refinement
+     */
+    void 
+    refine_mesh( std::shared_ptr<SimulationBase<dim>> base_in)
+    {
+      if (dim>1)
```

Comment:
```suggestion
      if (auto tria_ref = dynamic_cast<parallel::distributed::Triangulation<dim>>(base_in->triangulation))
```

The problem here is not if it is 1D or not, but if it a simple `Triangulation` or `p:d:T`.

Reply:
I've introduced a "dynamic_pointer_cast" at this point.